src/PhoneNumber/PrefixSelector/PrefixSelector.js

```diff
@@ -213,29 +224,36 @@ export class FlagSelector extends Component {
   get validOptions() {
     const { options } = this.props;
 
-    return options.filter(
-      ({ dialingCode, phonePattern }) =>
-        !isEmpty(dialingCode) && !isEmpty(phonePattern)
-    );
+    return options.filter(({ dialingCode }) => !isEmpty(dialingCode));
   }
 
-  render() {
-    const { value = '' } = this.props;
-    const { disabled, readOnly } = this.props;
-    const { isOpen, options, selectedIndex } = this.state;
+  renderOption = (option, index) => {
+    const { label, value, dialingCode } = option;
+    const { selectedIndex } = this.state;
+
+    const hasFocus = selectedIndex === index;
 
-    const optionList = options.map((option, i) => (
+    return (
       <Option
-        country={option.label}
-        dialingCode={option.dialingCode}
-        hasFocus={selectedIndex === i}
-        key={option.value}
+        country={label}
+        dialingCode={dialingCode}
+        hasFocus={hasFocus}
+        key={value}
         onClick={value => this.handleOptionSelected(value)}
         onMouseEnter={value => this.handleOptionHover(value)}
-        ref={option => this.setOptionRef(i, option)}
-        value={option.value}
+        onMouseOver={value => this.handleOptionHover(value)}
+        ref={option => this.setOptionRef(index, option)}
+        value={value}
       />
-    ));
+    );
+  };
+
+  render() {
+    const { value, disabled, readOnly } = this.props;
+    const { dialingCode, isOpen, options } = this.state;
+    const optionList = options.map(this.renderOption);
+    const countryPrefix =
+      dialingCode !== '' ? dialingCode : value !== '' ? value : false;
```

Comment:
This line is difficult to understand for me, and i don't see how is it tested

Reply:
Fixed